libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/reverse.hpp

```diff
@@ -113,28 +346,298 @@ namespace hpx { namespace parallel { inline namespace v1 {
     /// within each thread.
     ///
     /// \returns  The \a reverse_copy algorithm returns a
-    ///           \a hpx::future<in_out_result<BidirIter, OutIter> >
+    ///           \a hpx::future<ranges::reverse_copy_result<
+    ///            typename hpx::traits::range_iterator<Rng>::type, OutIter>>
     ///           if the execution policy is of type
     ///           \a sequenced_task_policy or
     ///           \a parallel_task_policy and
-    ///           returns \a in_out_result<BidirIter, OutIter>
+    ///           returns \a ranges::reverse_copy_result<
+    ///            typename hpx::traits::range_iterator<Rng>::type, OutIter>
     ///           otherwise.
-    ///           The \a copy algorithm returns the pair of the input iterator
-    ///           forwarded to the first element after the last in the input
-    ///           sequence and the output iterator to the
-    ///           element in the destination range, one past the last element
-    ///           copied.
+    ///           The \a reverse_copy algorithm returns
+    ///           an object equal to {last, result + N} where N = last - first
     ///
+    template <typename ExPolicy, typename Rng, typename OutIter>
+    typename util::detail::algorithm_result<ExPolicy,
+        ranges::reverse_copy_result<
+            typename hpx::traits::range_iterator<Rng>::type, OutIter>>::type
+    reverse_copy(ExPolicy&& policy, Rng&& rng, OutIter result);
+}}    // namespace hpx::ranges
+
+#else    // DOXYGEN
+
+#include <hpx/config.hpp>
+#include <hpx/concepts/concepts.hpp>
+#include <hpx/iterator_support/range.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/iterator_support/traits/is_range.hpp>
+#include <hpx/parallel/util/result_types.hpp>
+
+#include <hpx/algorithms/traits/projected_range.hpp>
+#include <hpx/parallel/algorithms/reverse.hpp>
+#include <hpx/parallel/tagspec.hpp>
+#include <hpx/parallel/util/projection_identity.hpp>
+
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace parallel { inline namespace v1 {
+
+    // clang-format off
+    template <typename ExPolicy, typename Rng,
+        HPX_CONCEPT_REQUIRES_(
+            hpx::is_execution_policy<ExPolicy>::value &&
+            hpx::traits::is_range<Rng>::value)>
+    // clang-format on
+    HPX_DEPRECATED_V(1, 7,
+        "hpx::parallel::reverse is deprecated, use hpx::ranges::reverse "
+        "instead") typename util::detail::algorithm_result<ExPolicy,
+        typename hpx::traits::range_iterator<Rng>::type>::type
+        reverse(ExPolicy&& policy, Rng&& rng)
+    {
+        typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
+
+        return detail::reverse<
+            typename hpx::traits::range_iterator<Rng>::type>()
+            .call(std::forward<ExPolicy>(policy), is_seq(),
+                hpx::util::begin(rng), hpx::util::end(rng));
```

Comment:
This will most likely not compile anymore (after the changes on master). You need to remove the `is_seq` predicate altogether.

Reply:
done